test/e2e/resource_placement_rollout_test.go

```diff
@@ -0,0 +1,1193 @@
+
+			rpStatusUpdatedActual := customizedPlacementStatusUpdatedActual(rpKey, wantSelectedResources, []string{memberCluster1EastProdName}, nil, observedResourceIdx, false)
+			Eventually(rpStatusUpdatedActual, 4*time.Duration(unAvailablePeriodSeconds)*time.Second, eventuallyInterval).Should(Succeed(), "Failed to update RP status as expected")
+		})
+
+		It("delete the RP and related resources", func() {
+		})
+	})
+
+	Context("Test an RP place custom resource successfully, should wait to update resource on multiple member clusters", Ordered, func() {
+		workNamespace := appNamespace()
+		var wantSelectedResources []placementv1beta1.ResourceIdentifier
+		var rp *placementv1beta1.ResourcePlacement
+		unAvailablePeriodSeconds := 30
+		var observedResourceIdx string
+
+		BeforeAll(func() {
+			// Create the test resources.
+			wantSelectedResources = []placementv1beta1.ResourceIdentifier{
+				{
+					Group:     testv1alpha1.GroupVersion.Group,
+					Kind:      testCustomResourceKind,
+					Name:      testCustomResource.Name,
+					Version:   testv1alpha1.GroupVersion.Version,
+					Namespace: workNamespace.Name,
+				},
+			}
+		})
+
+		It("create the RP that select the custom resource", func() {
+			rp = &placementv1beta1.ResourcePlacement{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       rpName,
+					Namespace:  workNamespace.Name,
+					Finalizers: []string{customDeletionBlockerFinalizer},
+				},
+				Spec: placementv1beta1.PlacementSpec{
+					ResourceSelectors: []placementv1beta1.ResourceSelectorTerm{
+						{
+							Group:   testv1alpha1.GroupVersion.Group,
+							Kind:    testCustomResourceKind,
+							Version: testv1alpha1.GroupVersion.Version,
+							Name:    testCustomResource.Name,
+						},
+					},
+					Policy: &placementv1beta1.PlacementPolicy{
+						PlacementType: placementv1beta1.PickAllPlacementType,
+					},
+					Strategy: placementv1beta1.RolloutStrategy{
+						Type: placementv1beta1.RollingUpdateRolloutStrategyType,
+						RollingUpdate: &placementv1beta1.RollingUpdateConfig{
+							MaxUnavailable: &intstr.IntOrString{
+								Type:   intstr.Int,
+								IntVal: 1,
+							},
+							UnavailablePeriodSeconds: ptr.To(unAvailablePeriodSeconds),
+						},
+					},
+				},
+			}
+			Expect(hubClient.Create(ctx, rp)).To(Succeed(), "Failed to create RP")
+		})
+
+		It("should update RP status as expected", func() {
+			Eventually(func() error {
+				rp := &placementv1beta1.ResourcePlacement{}
+				if err := hubClient.Get(ctx, rpKey, rp); err != nil {
+					return fmt.Errorf("failed to get RP: %w", err)
+				}
+
+				if diff := cmp.Diff(rp.Status.SelectedResources, wantSelectedResources, cmpopts.SortSlices(utils.LessFuncResourceIdentifier)); diff != "" {
+					return fmt.Errorf("selected resources mismatched (-got, +want): %s", diff)
+				}
+				// Use the fresh observed resource index to verify the RP status later.
```

Comment:
just wondering why the observedResourceIndex is not predictable, should it be 0?

Reply:
Since this is a custom resource, and our resource discovery has a 15s period, there's chance that even the resources are already created at the beginning of the test, the custom resource has not been detected at the time when rp gets created. So it may take 2 resourceSnapshots: the 1st one is empty without the custom resource, then 2nd one has it.